modules/Kconfig.s32

```diff
@@ -0,0 +1,29 @@
+# NXP S32 MCUs family
```

Comment:
Why is this not part of the modules/Kconfig.mcux

Reply:
The NXP S32 family is not covered by MCUXpresso SDK

Reply:
But you are using the same HAL repo for it.

Reply:
> While this is all NXP, we have different business units and product lines. For the moment, the internal guidance has been to separate this so I'm okay with a Kconfig.s32.
> 
> We can consider merging these into a single Kconfig.nxp in a separate PR from this PR.

I don't know if we have examples of how best to handle this pattern - but feels like we should have a 1:1 correspondence with the 'module'.  So if S32 lives in the existing hal_nxp module, than it should be treated as one thing.  @carlescufi do we have cases of different module/Kconfigs that map to one module repo (at this level of the hierarchy)?

Reply:
maybe create `modules/hal_nxp` and multiple `Kconfig.*` inside include by a main `Kconfig`?

Reply:
As there is a pattern with Kconfig.imx this is ok for now.  However, I agree with @gmarull that all that should get put under a `hal_nxp` dir.  Otherwise its very difficult to know which Kconfig files are associated with which modules.

Reply:
Ok, we will consider this suggestion and address it in a separate PR then.